rclpy/rclpy/node.py

```diff
@@ -210,6 +210,11 @@ def executor(self, new_executor: Executor) -> None:
             new_executor.add_node(self)
             self.__executor_weakref = weakref.ref(new_executor)
 
+    def _wake_executor(self):
+        executor = self.executor
```

Comment:
This seems unnecessary, is there a reason for it?

Reply:
vs

```python3
if self.executor:
   self.executor.wake()
```

`self.executor` is a property that is set to `None` when the node is removed from an executor. If the node is removed after the check then the call to `wake` might raise `AttributeError`